### DISCUSSION TITLE
Best way to create a package to distribute for offline usage

### DISCUSSION THREAD

**Discussion by jogielis** [pdm-project/pdm#2486] [Q&A] [answered]
What would be a good way to create a package or bundle to be able to allow offline installation?
With offline I mean that pip repositories will not be available, all pip packages should be contained in the package. And PDM should be able to find all necessary packages offline during installation of the virtual environment.

**Comment by frostming** [comment, marked as answer]
I made this in 30 minutes: https://github.com/pdm-project/pdm-download

**Reply by jogielis** [reply]
Thanks for the quick response and solution, I will try it out.
